invenio_cli/commands/translations.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-Cli is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Invenio module to ease the creation and management of applications."""
+
+from pathlib import Path
+
+from ..commands import Commands
+from ..helpers.filesystem import symlink_project_file_or_folder
+from .steps import CommandStep, FunctionStep
+
+
+class TranslationsCommands(Commands):
+    """Translations CLI commands."""
+
+    CMD_PREFIX = ["pipenv", "run"]
+
+    @classmethod
+    def extract(
+        cls,
+        babel_file,
+        output_file,
+        input_dirs,
+        msgid_bugs_address,
+        copyright_holder,
+        add_comments="NOTE",
+    ):
+        """Extract messages from source code and templates."""
+
+        cmd = cls.CMD_PREFIX + [
+            "pybabel",
+            "extract",
+            f"--mapping-file={babel_file}",
+            f"--output-file={output_file}",
+            f"--input-dirs={input_dirs}",
+            f"--msgid-bugs-address={msgid_bugs_address}",
+            f"--copyright-holder={copyright_holder}",
+            f"--add-comments={add_comments}",
+        ]
+
+        return [
+            CommandStep(
+                cmd=cmd,
+                env={"PIPENV_VERBOSITY": "-1"},
+                message=f"Extracting i18n messages from {input_dirs}...",
+            )
+        ]
+
+    @classmethod
+    def init(cls, output_dir, input_file, locale):
+        """initialize a new language catalog."""
+
+        cmd = cls.CMD_PREFIX + [
+            "pybabel",
+            "init",
+            f"--output-dir={output_dir}",
+            f"--input-file={input_file}",
+            f"--locale={locale}",
+        ]
+
+        return [
+            CommandStep(
+                cmd=cmd,
+                env={"PIPENV_VERBOSITY": "-1"},
+                message=f"Initializing message catalog for {locale}...",
+            )
+        ]
+
+    @classmethod
+    def update(cls, output_dir, input_file):
+        """Update the message catalog."""
+
+        cmd = cls.CMD_PREFIX + [
```

Comment:
Seems to be almost equal to `init`. However, I do not find a semantically meaningful way of just sharing the cmd. It would make things ad-hoc... so duplication is known evil here.